From 1c564133f38669e52dd3220885e71e01d2c2ec12 Mon Sep 17 00:00:00 2001
From: Turo Soisenniemi <turo.soisenniemi@outlook.com>
Date: Wed, 14 Feb 2018 19:41:13 +0000
Subject: [PATCH 13/25] Set Asciidoctor outfilesuffix to default .adoc

Without setting outfilesuffix Asciidoctor defaults to .html. When .html is used,
inter-document cross references do not work in Gitlab's repository / file renderer.
See more information from
http://asciidoctor.org/docs/user-manual/#navigating-between-source-files
and
http://asciidoctor.org/docs/user-manual/#inter-document-cross-references
---
 .../unreleased/asciidoc_inter_document_cross_references.yml       | 5 +++++
 lib/gitlab/asciidoc.rb                                            | 3 ++-
 spec/lib/gitlab/asciidoc_spec.rb                                  | 8 ++++++++
 3 files changed, 15 insertions(+), 1 deletion(-)
 create mode 100644 changelogs/unreleased/asciidoc_inter_document_cross_references.yml

diff --git a/changelogs/unreleased/asciidoc_inter_document_cross_references.yml b/changelogs/unreleased/asciidoc_inter_document_cross_references.yml
new file mode 100644
index 0000000..34b2675
--- /dev/null
+++ b/changelogs/unreleased/asciidoc_inter_document_cross_references.yml
@@ -0,0 +1,5 @@
+---
+title: Asciidoc now support inter-document cross references between files in repository
+merge_request: 17125
+author: Turo Soisenniemi
+type: changed
diff --git a/lib/gitlab/asciidoc.rb b/lib/gitlab/asciidoc.rb
index cead1c7..11a1230 100644
--- a/lib/gitlab/asciidoc.rb
+++ b/lib/gitlab/asciidoc.rb
@@ -10,7 +10,8 @@ module Gitlab
 
     DEFAULT_ADOC_ATTRS = [
       'showtitle', 'idprefix=user-content-', 'idseparator=-', 'env=gitlab',
-      'env-gitlab', 'source-highlighter=html-pipeline', 'icons=font'
+      'env-gitlab', 'source-highlighter=html-pipeline', 'icons=font',
+      'outfilesuffix=.adoc'
     ].freeze
 
     # Public: Converts the provided Asciidoc markup into HTML.
diff --git a/spec/lib/gitlab/asciidoc_spec.rb b/spec/lib/gitlab/asciidoc_spec.rb
index f668f78..2a0e19a 100644
--- a/spec/lib/gitlab/asciidoc_spec.rb
+++ b/spec/lib/gitlab/asciidoc_spec.rb
@@ -95,6 +95,14 @@ module Gitlab
           expect(render(input, context)).to include('<p><code data-math-style="inline" class="code math js-render-math">2+2</code> is 4</p>')
         end
       end
+
+      context 'outfilesuffix' do
+        it 'defaults to adoc' do
+          output = render("Inter-document reference <<README.adoc#>>", context)
+
+          expect(output).to include("a href=\"README.adoc\"")
+        end
+      end
     end
 
     def render(*args)
-- 
2.10.1 (Apple Git-78)

